HousingManagementSystemApi/Controllers/AddressesController.cs

```diff
@@ -12,24 +14,28 @@ namespace HousingManagementSystemApi.Controllers
     [ApiVersion("1.0")]
     public class AddressesController : ControllerBase
     {
-        private readonly IRetrieveAddressesUseCase retrieveAddressesUseCase;
+        private readonly IRetrieveAddressesUseCase retrieveAddressesUseCase;
+        private readonly ILogger<AddressesController> _logger;
 
-        public AddressesController(IRetrieveAddressesUseCase retrieveAddressesUseCase)
+        public AddressesController(IRetrieveAddressesUseCase retrieveAddressesUseCase, ILogger<AddressesController> logger)
         {
             this.retrieveAddressesUseCase = retrieveAddressesUseCase;
+            _logger = logger;
         }
 
         [HttpGet]
         public async Task<IActionResult> Address([FromQuery] string postcode)
         {
             try
             {
-                var result = await retrieveAddressesUseCase.Execute(postcode);
+                var result = await retrieveAddressesUseCase.Execute(postcode);
+                _logger.LogInformation($"Number of results being returned by AddressesController.Address: {result.Count()} - for postcode ${postcode}");
```

Comment:
I would suggest you use `result?.Count()` instead. You don't want the log to throw an error itself if the result is null

Reply:
Ive made that mistake a few times

Reply:
Also, there is an additional $ in front of the postCode

Reply:
Good suggestion, will edit, thanks